dockstore-webservice/src/main/java/io/dockstore/webservice/languages/CWLHandler.java

```diff
@@ -725,28 +725,32 @@ public VersionTypeValidation validateWorkflowSet(Set<SourceFile> sourcefiles, St
         Optional<SourceFile> mainDescriptor = filteredSourcefiles.stream().filter((sourceFile -> Objects.equals(sourceFile.getPath(), primaryDescriptorFilePath))).findFirst();
 
         boolean isValid = true;
-        String validationMessage = null;
+        StringBuilder validationMessage = new StringBuilder();
         Map<String, String> validationMessageObject = new HashMap<>();
 
         if (mainDescriptor.isPresent()) {
             Yaml yaml = new Yaml();
             String content = mainDescriptor.get().getContent();
             if (content == null || content.isEmpty()) {
                 isValid = false;
-                validationMessage = "Primary descriptor is empty.";
+                validationMessage.append("Primary descriptor is empty.");
             } else if (!content.contains("class: Workflow")) {
                 isValid = false;
-                validationMessage = "Requires class: Workflow.";
+                validationMessage.append("A CWL workflow requires 'class: Workflow'.");
+                if (content.contains("class: CommandLineTool") || content.contains("class: ExpressionTool")) {
+                    String cwlClass = content.contains("class: CommandLineTool") ? "CommandLineTool" : "ExpressionTool";
+                    validationMessage.append(" This file contains 'class: ").append(cwlClass).append("'. Did you mean to register a tool?");
```

Comment:
I prefer: `validationMessage.append(String.format(" This file contains 'class: %s'. Did you mean to register a tool?", cwlClass));`

Reply:
String.format is less efficient than StringBuilder so I just stuck with using StringBuilder's append since using String.format would defeat the purpose of using StringBuilder for efficiency.

Reply:
It's for readability: https://stackoverflow.com/a/44127617